homeassistant/components/roborock/config_flow.py

```diff
@@ -91,6 +100,19 @@ async def async_step_code(
                 _LOGGER.exception(ex)
                 errors["base"] = "unknown"
             else:
+                if self.source == SOURCE_REAUTH:
+                    assert self.reauth_entry is not None
```

Comment:
Could save one assert with:
```suggestion
                if self.reauth_entry is not None:
```